src/components/synchronization-page/OrganisationUnitPage.jsx

```diff
@@ -1,25 +1,70 @@
 import React from "react";
 import PropTypes from "prop-types";
 import i18n from "@dhis2/d2-i18n";
+import _ from "lodash";
+
 import {
     OrganisationUnitGroupModel,
     OrganisationUnitGroupSetModel,
     OrganisationUnitModel,
 } from "../../models/d2Model";
+import { getOrgUnitSubtree } from "../../logic/metadata";
 import GenericSynchronizationPage from "./GenericSynchronizationPage";
 
 export default class OrganisationUnitPage extends React.Component {
     static propTypes = {
         d2: PropTypes.object.isRequired,
     };
 
+    state = {
+        children: [],
+        metadataTableKey: Math.random(),
+    };
+
     models = [OrganisationUnitModel, OrganisationUnitGroupModel, OrganisationUnitGroupSetModel];
 
+    clearChildren = () => this.setState({ children: [] });
+
+    selectChildren = async selectedOUs => {
+        const { d2 } = this.props;
+        const children = _.flatten(
+            await Promise.all(selectedOUs.map(selectedOU => getOrgUnitSubtree(d2, selectedOU.id)))
```

Comment:
Risky unbounded parallel requests, specially when you say that `getOrgUnitSubtree` is a "very dangerous operation"  that tends to crash the server :). I keep forgetting if we have a promise pool in this project.

Reply:
Yeah, initially the action was not multiple. I will replace the ```promise.all``` with a ```for of async```.

I faced the crashing of the server several times with that request on 2.28 WHO servers a year ago. We had to completely deprecate that call at the time but now on 2.30 appears to work fine, even though it is still a very demanding request. I prefer to have the warning in the code, just in case!